### PR TITLE
Fix model preparation for custom signatures

### PR DESCRIPTION
**Pull request by Bogdan-Wiederspan**
Fixes a minor bug in the graph preparation code that prevented models from being compiled if they were saved with a signature other than the default (serving_default). 
The relevant test has also been updated to cover this edge case, ensuring models with custom signatures compile as expected.

### CODE REVIEW

File: tests/test_compile_tf_graph.py [hunk 1]
```diff
@@ -58,59 +58,73 @@ def test_compile_tf_graph_static_preparation(self):
...
+            spec = [
+                tf.TensorSpec(shape=(None, 2), dtype=tf.float32, name="inputs"),
+                tf.TensorSpec(shape=(None, 3), dtype=tf.float32, name="inputs_1"),
+                tf.TensorSpec(shape=(None, 10), dtype=tf.float32, name="inputs_2"),
+                ]
+
+            conc_func = tf.function(model.call).get_concrete_function(spec)
+            signatures = {
+                tf.saved_model.DEFAULT_SERVING_SIGNATURE_DEF_KEY: conc_func,
+                "custom_signature":conc_func,
```
Comment (riga) [inline tests/test_compile_tf_graph.py]:
```suggestion
                "custom_signature": conc_func,
```

**Review by riga — CHANGES_REQUESTED** [review]
